File: mobile/types/practice.ts
export type Difficulty = "beginner" | "intermediate" | "advanced";

export type PracticeStatus = "pending" | "completed";

export type Practice = {
  id: number;
  title: string;
  description: string;
  duration: number;
  difficulty: Difficulty;
  status: PracticeStatus;
  created_at?: string;
  updated_at?: string;
};

export type CreatePracticeInput = {
  title: string;
  description: string;
  duration: number;
  difficulty: Difficulty;
};

export type UpdatePracticeInput = {
  title?: string;
  description?: string;
  duration?: number;
  difficulty?: Difficulty;
  status?: PracticeStatus;
};

File: backend/app/models/practice.py
from datetime import datetime
from enum import Enum

from sqlalchemy import DateTime, Integer, String, Text
from sqlalchemy import Enum as SQLEnum
from sqlalchemy.orm import Mapped, mapped_column

from ..database import Base


class Difficulty(str, Enum):
    BEGINNER = "beginner"
    INTERMEDIATE = "intermediate"
    ADVANCED = "advanced"


class PracticeStatus(str, Enum):
    PENDING = "pending"
    COMPLETED = "completed"


class Practice(Base):
    __tablename__ = "practices"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    title: Mapped[str] = mapped_column(String(200), nullable=True)
    description: Mapped[str] = mapped_column(Text, nullable=True)
    duration: Mapped[int] = mapped_column(Integer, nullable=False)
    difficulty: Mapped[Difficulty] = mapped_column(
        SQLEnum(Difficulty),
        nullable=False,
    )
    status: Mapped[PracticeStatus] = mapped_column(
        SQLEnum(PracticeStatus), nullable=False
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.now,
        nullable=False,
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.now,
        onupdate=datetime.now,
        nullable=False,
    )
